src/osd/OSD.cc

```diff
@@ -7149,6 +7159,12 @@ void OSD::sched_scrub()
       PGRef pg = _lookup_lock_pg(scrub.pgid);
       if (!pg)
 	continue;
+      if (allow_requested_repair_only && !pg->scrubber.must_repair) {
+        dout(10) << __func__ << " skip " << scrub.pgid
+                 << " because repairing is not explicitly requested on it"
+                 << dendl;
+        continue;
```

Comment:
Need to unlock the pg here before "continue"

Reply:
Thanks a lot for the good catch! I will address it.

Reply:
Done

Reply:
I don't see the unlock here.